src/test/scala/io/iohk/ethereum/consensus/validators/BlockHeaderValidatorSpec.scala

```diff
@@ -143,61 +151,93 @@ class BlockHeaderValidatorSpec extends FlatSpec with Matchers with PropertyCheck
-    val res = blockHeaderValidator.validate(pausedDifficultyBombBlock, pausedDifficultyBombBlockParent)
+    val parentBody: BlockBody = BlockBody.empty
+    val parent = Block(pausedDifficultyBombBlockParent, parentBody)
+
+    val res = blockHeaderValidator.validate(pausedDifficultyBombBlock, parent)
     res shouldBe Right(BlockHeaderValid)
   }
 
-  it should "properly calculate the difficulty after difficulty bomb resume" in new EphemBlockchainTestSetup {
-    val parentHeader: BlockHeader = validBlockParent.copy(
+  it should "properly calculate the difficulty after difficulty bomb resume (with reward reduction)" in new EphemBlockchainTestSetup {
+    val parentHeader: BlockHeader = validParentBlockHeader.copy(
       number = 5000101,
       unixTimestamp = 1513175023,
       difficulty = BigInt("22627021745803"))
+    val parent = Block(parentHeader, parentBody)
 
     val blockNumber: BigInt = parentHeader.number + 1
     val blockTimestamp: Long = parentHeader.unixTimestamp + 6
 
-    val difficulty: BigInt = difficultyCalculator.calculateDifficulty(blockNumber, blockTimestamp, parentHeader)
-    val expected = BigInt("22638338531720")
+    val difficulty: BigInt = difficultyCalculator.calculateDifficulty(blockNumber, blockTimestamp, parent)
+    val expected = BigInt("22638070358408")
 
     difficulty shouldBe expected
   }
 
   it should "properly calculate the difficulty after difficulty defuse" in new EphemBlockchainTestSetup {
-    val parentHeader: BlockHeader = validBlockParent.copy(
+    val parentHeader: BlockHeader = validParentBlockHeader.copy(
       number = 5899999,
       unixTimestamp = 1525176000,
       difficulty = BigInt("22627021745803"))
+    val parent = Block(parentHeader, parentBody)
 
     val blockNumber: BigInt = parentHeader.number + 1
     val blockTimestamp: Long = parentHeader.unixTimestamp + 6
 
-    val difficulty: BigInt = difficultyCalculator.calculateDifficulty(blockNumber, blockTimestamp, parentHeader)
+    val difficulty: BigInt = difficultyCalculator.calculateDifficulty(blockNumber, blockTimestamp, parent)
     val blockDifficultyWihtoutBomb = BigInt("22638070096264")
 
     difficulty shouldBe blockDifficultyWihtoutBomb
   }
 
+  it should "properly calculate a block after block reward reduction (without uncles)" in new EphemBlockchainTestSetup {
+    val parent = Block(afterRewardReductionBlockHeader, parentBody)
+
+    val blockNumber: BigInt = afterRewardReductionParentBlockHeader.number
+    val blockTimestamp: Long = afterRewardReductionParentBlockHeader.unixTimestamp
+
+    val difficulty: BigInt = difficultyCalculator.calculateDifficulty(blockNumber, blockTimestamp, parent)
+    val blockDifficultyAfterRewardReduction = BigInt("3484099629090779")
+
+    difficulty shouldBe blockDifficultyAfterRewardReduction
+  }
+
+  it should "properly validate the difficulty after block reward reduction (without uncles)" in new EphemBlockchainTestSetup {
```

Comment:
This test is failing becouse of Pow validation error ( it is probably because of what you told me, that we do not have  proper mix hash). I think we can delete that test as we only care about difficulty ( test above)